### PR TITLE
feat(Core/Guild): Allow to limit the amount of members within a guild

### PR DESCRIPTION
**Pull request by Kitzunu**
## Changes Proposed:

This PR proposes changes to:
-  [ ] Core (units, players, creatures, game systems).
-  [ ] Scripts (bosses, spell scripts, creature scripts).
-  [ ] Database (SAI, creatures, etc).

## Issues Addressed:

- Closes 

## SOURCE:

The changes have been validated through:
- [ ] Live research (checked on live servers, e.g Classic WotLK, Retail, etc.)
- [ ] Sniffs (remember to share them with the open source community!)
- [ ] Video evidence, knowledge databases or other public sources (e.g forums, Wowhead, etc.)
- [ ] The changes promoted by this pull request come partially or entirely from another project (cherry-pick). **Cherry-picks must be committed using the proper --author tag in order to be accepted, thus crediting the original authors, unless otherwise unable to be found**

## Tests Performed:

This PR has been:
- [ ] Tested in-game by the author.
- [ ] Tested in-game by other community members/someone else other than the author/has been live on production servers.
- [ ] This pull request requires further testing and may have edge cases to be tested.

## How to Test the Changes:

- [ ] This pull request can be tested by following the reproduction steps provided in the linked issue
- [ ] This pull request requires further testing. Provide steps to test your changes. If it requires any specific setup e.g multiple players please specify it as well.

1.
2.
3.

## Known Issues and TODO List:

- [ ]
- [ ]

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

File: src/server/game/Guilds/Guild.cpp [hunk 1]
```diff
@@ -1440,6 +1440,14 @@ void Guild::HandleInviteMember(WorldSession* session, std::string const& name)
     if (pInvitee->GetSocial()->HasIgnore(player->GetGUID()))
         return;
 
+    uint32 memberLimit = sConfigMgr->GetOption<uint32>("Guild.MemberLimit", 0);
+    if (memberLimit > 0 && player->GetGuild()->GetMemberCount() >= memberLimit)
+    {
+        player->SendSystemMessage("Guild has the maximum amount of members. Cannot send another invite.");
```
Comment (Nefertumm) [inline src/server/game/Guilds/Guild.cpp:1446]: Maybe we could send the maximum amount within the message?

Comment (Kitzunu) [reply src/server/game/Guilds/Guild.cpp:1446]: I thought about it. Tested it and it didnt work. But maybe SendSystemMessage dont support fmt style?

Comment (Nefertumm) [reply src/server/game/Guilds/Guild.cpp:1446]: Nopers, you need to do as before, with %u

**Review by elthehablo — APPROVED** [review]

File: src/server/game/Guilds/Guild.cpp (hunk 1 above)
Comment (Nyeriah) [reply src/server/game/Guilds/Guild.cpp:1446]: Its PSendSystemMessage to send variables

Comment (Nyeriah) [reply src/server/game/Guilds/Guild.cpp:1446]: And its not supported by the player helper I added, so need to build it with the chathandler

Comment (Kitzunu) [reply src/server/game/Guilds/Guild.cpp:1446]: Making my life difficult D:

I will fix this before I merge

Comment (Kitzunu) [reply src/server/game/Guilds/Guild.cpp:1446]: Maybe we should expose it to player helper 🤷

Comment (Kitzunu) [reply src/server/game/Guilds/Guild.cpp]: image
